dds/DCPS/ValueCommon.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_VALUE_COMMON_H
+#define OPENDDS_DCPS_VALUE_COMMON_H
+
+#include "XTypes/TypeObject.h"
+
+#include <dds/Versioned_Namespace.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class EnumHelper {
+public:
+  virtual ~EnumHelper() {}
+  virtual bool get_value(ACE_CDR::Long& value, const char* name) const = 0;
+  virtual bool get_name(const char*& name, ACE_CDR::Long value) const = 0;
```

Comment:
Since we're going to have `optional` available at some point in the not-too-distant future (either from `std::` or our own), this would be a good use of it.
Doesn't need to be in this PR.

Reply:
Can you elaborate on how `optional` can be used here?